pkg/addons/ensure.go

```diff
@@ -35,27 +35,28 @@ var (
 	// embeddedAddons is a list of addons that are embedded in the KubeOne
 	// binary. Those addons are skipped when applying a user-provided addon with the same name.
 	embeddedAddons = map[string]string{
-		resources.AddonCCMAws:             "",
-		resources.AddonCCMAzure:           "",
-		resources.AddonCCMDigitalOcean:    "",
-		resources.AddonCCMHetzner:         "",
-		resources.AddonCCMOpenStack:       "",
-		resources.AddonCCMEquinixMetal:    "",
-		resources.AddonCCMPacket:          "",
-		resources.AddonCCMVsphere:         "",
-		resources.AddonCNICanal:           "",
-		resources.AddonCNICilium:          "",
-		resources.AddonCNIWeavenet:        "",
-		resources.AddonCSIAwsEBS:          "",
-		resources.AddonCSIAzureDisk:       "",
-		resources.AddonCSIAzureFile:       "",
-		resources.AddonCSIDigitalOcean:    "",
-		resources.AddonCSIHetzner:         "",
-		resources.AddonCSIOpenStackCinder: "",
-		resources.AddonCSIVsphere:         "",
-		resources.AddonMachineController:  "",
-		resources.AddonMetricsServer:      "",
-		resources.AddonNodeLocalDNS:       "",
+		resources.AddonCCMAws:                 "",
+		resources.AddonCCMAzure:               "",
+		resources.AddonCCMDigitalOcean:        "",
+		resources.AddonCCMHetzner:             "",
+		resources.AddonCCMOpenStack:           "",
+		resources.AddonCCMEquinixMetal:        "",
+		resources.AddonCCMPacket:              "",
+		resources.AddonCCMVsphere:             "",
+		resources.AddonCNICanal:               "",
+		resources.AddonCNICilium:              "",
+		resources.AddonCNIWeavenet:            "",
+		resources.AddonCSIAwsEBS:              "",
+		resources.AddonCSIAzureDisk:           "",
+		resources.AddonCSIAzureFile:           "",
+		resources.AddonCSIDigitalOcean:        "",
+		resources.AddonCSIHetzner:             "",
+		resources.AddonCSIOpenStackCinder:     "",
+		resources.AddonCSIVsphere:             "",
+		resources.AddonMachineController:      "",
+		resources.AddonOperatingSystemManager: "",
```

Comment:
I don't think we need to add this here. Let's treat OSM as an "accessible" addon for the beginning, not an embedded addon.